### ISSUE TITLE
Shape Tensor must have type int32

### ISSUE THREAD

**Issue by anoushsepehri** [NVIDIA/TensorRT#618]
## Description

I am trying to convert a ONNX model to TensoRT engine using the following tutorial for the jetson nano with jetpack 4.4

https://devblogs.nvidia.com/speeding-up-deep-learning-inference-using-tensorflow-onnx-and-tensorrt/

I can succesfully convert the model to ONNX format however during the conversion to TensoRT, I get many errors. Mainly that I need atleast one output node, and that I am using Int64 weights. I created another post for the errors involving the output nodes, however I have been trying to find some hacks to get around it. Currently I included the following line of code in the engine.py document from the tutorial to mark the output node:

       network.mark_output(network.get_layer(network.num_layers - 1).get_output(0))

however I get the following errors when I try to convert:

[TensorRT] WARNING: onnx2trt_utils.cpp:217: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[TensorRT] WARNING: onnx2trt_utils.cpp:243: One or more weights outside the range of INT32 was clamped
[TensorRT] ERROR: Min__149:0: shape tensor must have type Int32.
[TensorRT] ERROR: Equal__430:0: shape tensor must have type Int32.
[TensorRT] ERROR: Equal__439:0: shape tensor must have type Int32.
[TensorRT] ERROR: Equal__622:0: shape tensor must have type Int32.
[TensorRT] ERROR: Equal__766:0: shape tensor must have type Int32.
[TensorRT] ERROR: Builder failed while analyzing shapes.

Any help would be much appreciated. The tensorflow models were trained and exported using the Object Detection API from tensorflow.
## Environment

**TensorRT Version**: 7.1.0 
**Operating System + Version**: Jetpack 4.4 on Jetson Nano
**Python Version (if applicable)**: 3.6.9
**TensorFlow Version (if applicable)**: 2.1.0 to train, 1.15.2 on Jetson Nano

**Comment by Water2style** [comment]
Same warning,WARNING: onnx2trt_utils.cpp:217: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[TensorRT] WARNING: onnx2trt_utils.cpp:243: One or more weights outside the range of INT32 was clamped

I use my pytorch model to ONNX....  Why it becomes INT64 after to ONNX

**Comment by ttyio** [comment]
Thanks for reporting

Hello @anoushsepehri , 
For the layers like `resize`, `shuffle`, `slice`, we have multiple inputs, and some of the input accept shape tensor only, the shape input must have type INT32, it seems the model might calculate the shape using elementwise operations (min, equal...), could you visualize your onnx to check if there is some fp32 activation/weights feed into these elementwise operations?

Hello @Water2style , it is the pytorch ONNX exporter, they export model in INT64, did you see accuracy issue in your inference?

**Comment by ttyio** [comment]
close since no activity for more than 3 weeks, please reopen if you still have issue, thanks!

**Comment by mikaellaine** [comment]
I was able to fix my version of this issue by doing a tf.clip_by_value() on indices when slicing tensors.
